java/test/jmri/jmrit/display/layoutEditor/LayoutEditorDialogsTest.java

```diff
@@ -0,0 +1,357 @@
+package jmri.jmrit.display.layoutEditor;
+
+import java.awt.GraphicsEnvironment;
+import java.awt.geom.Rectangle2D;
+import jmri.util.JUnitUtil;
+import jmri.util.junit.rules.RetryRule;
+import org.junit.*;
+import org.junit.rules.Timeout;
+import org.netbeans.jemmy.operators.JButtonOperator;
+import org.netbeans.jemmy.operators.JDialogOperator;
+import org.netbeans.jemmy.operators.JFrameOperator;
+import org.netbeans.jemmy.operators.JTextFieldOperator;
+
+/**
+ * Test simple functioning of LayoutEditorDialogs
+ *
+ * @author	Paul Bender Copyright (C) 2016
+ */
+public class LayoutEditorDialogsTest {
+
+    @Rule
+    public Timeout globalTimeout = Timeout.seconds(10); // 10 second timeout for methods in this test class.
+
+    @Rule    // allow 2 retries of intermittent tests
+    public RetryRule retryRule = new RetryRule(2); // allow 2 retries
+
+    /*
+     * This is called once before all tests
+     */
+    @BeforeClass
+    public static void setUpClass() {
+        JUnitUtil.setUp();
+        if (!GraphicsEnvironment.isHeadless()) {
+            layoutEditor = new LayoutEditor();
+            layoutEditorDialogs = layoutEditor.getLEDialogs();
+            layoutEditor.setPanelBounds(new Rectangle2D.Double(0, 0, 640, 480));
+        }
+    }
+
+    /*
+     * This is called once after all tests
+     */
+    @AfterClass
+    public static void tearDownClass() {
+        if (!GraphicsEnvironment.isHeadless()) {
+            JUnitUtil.dispose(layoutEditor);
+            layoutEditor = null;
+            layoutEditorDialogs = null;
+        }
+        JUnitUtil.tearDown();
+    }
+
+    private static LayoutEditor layoutEditor = null;
+    private static LayoutEditorDialogs layoutEditorDialogs = null;
+
+    @Test
+    public void testCtor() {
+        Assume.assumeFalse(GraphicsEnvironment.isHeadless());
+        Assert.assertNotNull("layoutEditor exists", layoutEditor);
+        Assert.assertNotNull("layoutEditorDialogs exists", layoutEditorDialogs);
+    }
+
+    @Test
+    public void testEnterGridSizes() {
+        Assume.assumeFalse(GraphicsEnvironment.isHeadless());
+
+        layoutEditorDialogs.enterGridSizes();
+        JFrameOperator jFrameOperator = new JFrameOperator(Bundle.getMessage("SetGridSizes"));
+        Assert.assertNotNull("jFrameOperator 1st exists", jFrameOperator);
```

Comment:
You don’t need to check to see if the Operator exists.  If the operator‘s constructor can’t find the frame, it will fail the test.

Reply:
Good to know… I’ll pull all of those.